src/analysis/special_functions/gamma/gamma.lean

```diff
@@ -0,0 +1,559 @@
+  { intro k, induction k with k hk,
+    { simp [Gamma],},
+    { rw [←hk, nat.succ_eq_add_one, ←add_assoc],
+      refine (Gamma_aux_recurrence2 s (⌊1 - s.re⌋₊ + k) _).symm,
+      rw nat.cast_add,
+      have i0 := nat.sub_one_lt_floor (1 - s.re),
+      simp only [sub_sub_cancel_left] at i0,
+      refine lt_add_of_lt_of_nonneg i0 _,
+      rw [←nat.cast_zero, nat.cast_le], exact nat.zero_le k, } },
+  convert (u $ n - ⌊1 - s.re⌋₊).symm, rw nat.add_sub_of_le,
+  by_cases (0 ≤ 1 - s.re),
+  { apply nat.le_of_lt_succ,
+    exact_mod_cast lt_of_le_of_lt (nat.floor_le h) (by linarith : 1 - s.re < n + 1) },
+  { rw nat.floor_of_nonpos, linarith, linarith },
+end
+
+/-- The recurrence relation for the `Γ` function. -/
+theorem Gamma_add_one (s : ℂ) (h2 : s ≠ 0) : Gamma (s+1) = s * Gamma s :=
+begin
+  let n := ⌊1 - s.re⌋₊,
+  have t1 : -s.re < n,
+  { simpa only [sub_sub_cancel_left] using nat.sub_one_lt_floor (1 - s.re) },
+  have t2 : -(s+1).re < n,
+  { rw [add_re, one_re], linarith, },
+  rw [Gamma_eq_Gamma_aux s n t1, Gamma_eq_Gamma_aux (s+1) n t2, Gamma_aux_recurrence1 s n t1],
+  field_simp, ring,
+end
+
+theorem Gamma_eq_integral {s : ℂ} (hs : 0 < s.re) : Gamma s = Gamma_integral s :=
+Gamma_eq_Gamma_aux s 0 (by { norm_cast, linarith })
+
+lemma Gamma_one : Gamma 1 = 1 :=
+by { rw Gamma_eq_integral, simpa using Gamma_integral_one, simp }
+
+theorem Gamma_nat_eq_factorial (n : ℕ) : Gamma (n+1) = n! :=
+begin
+  induction n with n hn,
+  { simpa using Gamma_one },
+  { rw (Gamma_add_one n.succ $ nat.cast_ne_zero.mpr $ nat.succ_ne_zero n),
+    simp only [nat.cast_succ, nat.factorial_succ, nat.cast_mul], congr, exact hn },
+end
+
+/-- At `0` the Gamma function is undefined; by convention we assign it the value `0`. -/
+lemma Gamma_zero : Gamma 0 = 0 :=
+by simp_rw [Gamma, zero_re, sub_zero, nat.floor_one, Gamma_aux, div_zero]
+
+/-- At `-n` for `n ∈ ℕ`, the Gamma function is undefined; by convention we assign it the value 0. -/
+lemma Gamma_neg_nat_eq_zero (n : ℕ) : Gamma (-n) = 0 :=
+begin
+  induction n with n IH,
+  { rw [nat.cast_zero, neg_zero, Gamma_zero] },
+  { have A : -(n.succ : ℂ) ≠ 0,
+    { rw [neg_ne_zero, nat.cast_ne_zero],
+      apply nat.succ_ne_zero },
+    have : -(n:ℂ) = -↑n.succ + 1, by simp,
+    rw [this, Gamma_add_one _ A] at IH,
+    contrapose! IH,
+    exact mul_ne_zero A IH }
+end
+
+lemma Gamma_conj (s : ℂ) : Gamma (conj s) = conj (Gamma s) :=
+begin
+  suffices : ∀ (n:ℕ) (s:ℂ) , Gamma_aux n (conj s) = conj (Gamma_aux n s), from this _ _,
+  intro n,
+  induction n with n IH,
+  { rw Gamma_aux, exact Gamma_integral_conj, },
+  { intro s,
+    rw Gamma_aux,
+    dsimp only,
+    rw [div_eq_mul_inv _ s, ring_hom.map_mul, conj_inv, ←div_eq_mul_inv],
+    suffices : conj s + 1 = conj (s + 1), by rw [this, IH],
+    rw [ring_hom.map_add, ring_hom.map_one] }
+end
+
+end Gamma_def
+
+/-! Now check that the `Γ` function is differentiable, wherever this makes sense. -/
+
+section Gamma_has_deriv
```

Comment:
This is the other code change: this section has been drastically shortened using the general results on Mellin transforms.